app/[addressOrDomain]/page.tsx

```diff
@@ -1,53 +1,142 @@
   const { address } = useAccount();
   const { starknetIdNavigator } = useContext(StarknetIdJsContext);
   const [initProfile, setInitProfile] = useState(false);
+  const [leaderboardData, setLeaderboardData] =
+    useState<LeaderboardToppersData>({
+      best_users: [],
+      total_users: 0,
+    });
   const [identity, setIdentity] = useState<Identity>();
   const [notFound, setNotFound] = useState(false);
   const [isOwner, setIsOwner] = useState(false);
+  const [quests, setQuests] = useState<QuestDocument[]>([]);
+  const [userRanking, setUserRanking] = useState<RankingData>({
+    first_elt_position: 0,
+    ranking: [],
+  });
   const dynamicRoute = usePathname();
-  const isMobile = useMediaQuery("(max-width:768px)");
-  const [achievements, setAchievements] = useState<BuildingsInfo[]>([]);
-  const [soloBuildings, setSoloBuildings] = useState<StarkscanNftProps[]>([]);
-  const sinceDate = useCreationDate(identity);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    if (!address) setIsOwner(false);
+  }, [address]);
+
+  const fetchCompletedQuests = useCallback(
+    async (addr: string) => {
+      try {
+        if (!addr) return;
+        const res = await getCompletedQuests(addr);
+
+        const updatedQuestsResults = await Promise.allSettled(
+          res.map((id: number) => getQuestById(id))
+        );
+        const successfulCompletedQuests = updatedQuestsResults
+          .filter(
+            (result): result is PromiseFulfilledResult<QuestDocument> =>
+              result.status === "fulfilled"
+          )
+          .map((result) => result.value);
+
+        setQuests(() => [
+          ...successfulCompletedQuests.filter((quest) => quest !== null),
+        ]);
+      } catch (err) {
+        console.log("Error while fetching quests", err);
+      }
+    },
+    [address, identity]
+  );
+
+  const fetchRanking = useCallback(
+    async (addr: string) => {
+      try {
+        if (!addr) return;
+        const res = await fetchLeaderboardRankings({
+          addr: hexToDecimal(addr),
+          page_size: 10,
+          shift: 0,
+          duration: "all",
+        });
+        if (!res) return;
+        setUserRanking(res);
+      } catch (err) {
+        console.log("Error while fetching leaderboard position", err);
+      }
+    },
+    [address]
+  );
+
+  const fetchLeaderboardData = useCallback(
+    async (addr: string) => {
+      try {
+        if (!addr) return;
+        const res = await fetchLeaderboardToppers({
+          addr: hexToDecimal(addr),
+          duration: "all",
+        });
+        if (!res) return;
+        setLeaderboardData(res);
+      } catch (err) {
+        console.log("Error while fetching leaderboard position", err);
+      }
+    },
+    [address]
+  );
+
+  const fetchPageData = useCallback(async (addr: string) => {
+    setLoading(true);
+    await fetchCompletedQuests(addr);
+    await fetchRanking(addr);
+    await fetchLeaderboardData(addr);
```

Comment:
For all these functions you're using the same `try` `catch` structure. 

You can create a util function like so to make sure that same structure is used everywhere without rewriting it.

Here is an example of such a function: 

```ts
const fetchData = async (address: string, fetchFunction: Function, onSuccess: Function) => {
  if (!address) return;
  try {
    const res = await fetchFunction();
    if (res) onSuccess(res);
  } catch (err) {
    console.lo("Error while fetching data", err);
  }
}
```

With this model, the `fetchRanking` function would look like so:

```ts
const fetchRanking = useCallback((addr: string) => {
  const getRanking = () => fetchLeaderboardRankings({ addr: hexToDecimal(addr), page_size: 10, shift: 0, duration: "all" });
  fetchData(addr, getRanking, setUserRanking);
}, [address]);

```

Reply:
Actually we already handle try catch inside the api calls itself, these ones are not needed here, i'll remove